code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm

```diff
@@ -0,0 +1,400 @@
+		return
+
+	..()
+	data_count += (amount*0.5)
+
+/mob/living/carbon/superior_animal/robot/greyson/true_boss_data_star/adjustFireLoss(amount)
+	if(amount >= 1000)
+		to_chat(world,"<b><font color='#ffaa00'>Anti-Cheat Shielding Successfully Deplyed. Data Collected.</font></b>")
+		built_up_rage += 1
+		if(built_up_rage > 2)
+			to_chat(world,"<b><font color='#ffaa00'>Lesser Being Orbitcal Beacon Shield Successfully Deplyed. Data Collected.</font></b>")
+			message_admins("<b><font color='#ffaa00'>Adminstation Being Data Collected.</font></b>")
+		return
+
+	..()
+	data_count += (amount*0.5)
+
+/mob/living/carbon/superior_animal/robot/greyson/true_boss_data_star/bullet_act(obj/item/projectile/proj)
+	..()
+
+	if(returning_fire)
+		returning_fire = FALSE
+		var/turf/proj_start_turf = proj.starting
+		for(var/obj in proj_start_turf.contents)
+			if(istype(obj, /obj/machinery/power/os_turret))
+				return		// Don't shoot other turrets
+		if(prob(data_count*0.01))
+			projectiletype = /obj/item/projectile/tether
+		if(prob(data_count*0.001))
+			projectiletype = /obj/item/projectile/bullet/c10x24
+		rapid = FALSE
+		OpenFire(proj_start_turf)
+		handled_mode()
+
+
+/mob/living/carbon/superior_animal/robot/greyson/true_boss_data_star/emp_act(severity)
+	data_count += rand(-100, 500)
+	if(data_count >= 3000)
+		to_chat(src,"<b><font color='#ffaa00'>Data Star EMP Shielding Deployed Successfully. Data Collected.</font></b>")
+		return
+	..()
+
+/mob/living/carbon/superior_animal/robot/greyson/true_boss_data_star/ex_act(severity)
+	data_count += rand(80, 100)
+	if(data_count >= 1000)
+		to_chat(src,"<b><font color='#ffaa00'>Data Star Bomb Shielding Deployed Successfully. Data Collected.</font></b>")
+		return
+	..()
+
+/mob/living/carbon/superior_animal/robot/greyson/true_boss_data_star/proc/try_n_build()
+	if(!able_to_build)
+		return
+	if(prob(1) && walls_to_make > 0)
+		new /obj/effect/window_lwall_spawn/smartspawn/onestar(src.loc)
+	if(prob(2))
+		new /obj/structure/grille(src.loc)
+	if(prob(3))
+		new /obj/random/mob/roomba/any(src.loc) //This could be a clean bot or a phazon, RnG!
+	if(turrets_can_build && turrets_to_make > 0)
+		if(prob(0.01*data_count))
+			turrets_to_make -= 1
+			new /obj/random/turret/os(src.loc)
+
+/mob/living/carbon/superior_animal/robot/greyson/true_boss_data_star/handle_stunned()
+	if(data_count >= 1200 && stunned)
+		SetParalysis(0)
+		SetStunned(0)
+		SetWeakened(0)
+		SetDrowsyness(0)
+		to_chat(src,"<b><font color='#ffaa00'>Data Star Anti-Stun Deployed Successfully. Data Collected.</font></b>")
+		return
+	else
+		data_count += 50 //We REALLY dislike being stunned
+		..()
+
+/mob/living/carbon/superior_animal/robot/greyson/true_boss_data_star/handle_weakened()
+	if(data_count >= 1200 && weakened)
+		SetParalysis(0)
+		SetStunned(0)
+		SetWeakened(0)
+		SetDrowsyness(0)
+		to_chat(src,"<b><font color='#ffaa00'>Data Star Anti-Weaken Deployed Successfully. Data Collected.</font></b>")
```

Comment:
Ditto this, don't just refer to code stuff in  character it's a bit...ehh. 

"Data Star Anti-Mobility Countermeasures Deployed Successfully. Data Collected."